### PR TITLE
Pull request to solve documentation problems.

### PR DESCRIPTION
**Pull request by eldy**

### CODE REVIEW

**Comment by rdoursenaud** [comment]
Doesn't look good. It derives the type from the parameters default value. Unfortunately, this will break type hinting for objects.

Please don't merge or cleanup manually before.

File: htdocs/comm/propal/class/propal.class.php [hunk 1]
```diff
@@ -308,7 +308,7 @@ function insert_discount($idremise)
      *      @param		string		$label				???
      *		@param      int			$date_start       	Start date of the line
      *		@param      int			$date_end         	End date of the line
-     *      @param		array		$array_option		extrafields array
+     *      @param		integer		$array_option		extrafields array
```
Comment (marcosgdf) [inline htdocs/comm/propal/class/propal.class.php:311]: ¿?

File: htdocs/comm/action/class/ical.class.php [hunk 2]
```diff
@@ -257,7 +257,7 @@ function retun_key_value($text)
      * Parse RRULE  return array
      *
      * @param 	string 	$value	string
-     * @return 	array
+     * @return 	string
```
Comment (marcosgdf) [inline htdocs/comm/action/class/ical.class.php:260]: ¿?

File: htdocs/comm/propal/class/propal.class.php [hunk 3]
```diff
@@ -480,7 +480,7 @@ function addline($desc, $pu_ht, $qty, $txtva, $txlocaltax1=0, $txlocaltax2=0, $f
      *  @param		int			$type				0/1=Product/service
      *	@param      int			$date_start       	Start date of the line
      *	@param      int			$date_end         	End date of the line
-	 *  @param		array		$array_option		extrafields array
+	 *  @param		integer		$array_option		extrafields array
```
Comment (marcosgdf) [inline htdocs/comm/propal/class/propal.class.php:483]: ¿?

**Comment by marcosgdf** [comment]
I see it breaks most PHP documentation

**Comment by rdoursenaud** [comment]
Keep in mind that the root problem is that our default values are not good. They should be of the expected type.
If we want to continue accepting legacy values, let's say an user object and 0 we should document the type as "User|int".

My 2 cents

**Pull request closed without merging**